tests/GridFS/StreamWrapperFunctionalTest.php

```diff
@@ -169,6 +169,29 @@ public function testWritableStreamStat()
         $stat = fstat($stream);
         $this->assertSame(6, $stat[7]);
         $this->assertSame(6, $stat['size']);
+  }
+
+    public function testWritableStreamStatAfterSaving()
+    {
+        $id = new ObjectId;
+        $stream = $this->bucket->openUploadStream('filename', ['chunkSizeBytes' => 1024, '_id' => $id]);
+
+        $this->assertSame(6, fwrite($stream, 'foobar'));
+        $this->assertTrue(fclose($stream));
+        $currentTimestamp = time();
+        $stream = $this->bucket->openDownloadStream($id);
+
+        $stat = fstat($stream);
+        $this->assertSame(0100444, $stat[2]);
+        $this->assertSame(0100444, $stat['mode']);
+        $this->assertSame(6, $stat[7]);
+        $this->assertSame(6, $stat['size']);
+        $this->assertGreaterThanOrEqual($currentTimestamp, $stat[9]);
+        $this->assertGreaterThanOrEqual($currentTimestamp, $stat['mtime']);
+        $this->assertGreaterThanOrEqual($currentTimestamp, $stat[10]);
+        $this->assertGreaterThanOrEqual($currentTimestamp, $stat['ctime']);
+        $this->assertSame(1024, $stat[11]);
+        $this->assertSame(1024, $stat['blksize']);
```

Comment:
Note that this field reports `chunkSize`, so it would become `4` if you use `length-10`.

https://github.com/mongodb/mongo-php-library/blob/f4801b8086cc2cf67997ea9e2bad671ff87d3ff1/src/GridFS/StreamWrapper.php#L197-L199